File: backend/src/main/java/com/scmc/domain/dto/DecryptResponse.java
package com.scmc.domain.dto;

import com.scmc.domain.dto.audit.AuditStep;
import java.util.List;

public record DecryptResponse (

    String encryptedMessage,

    String permutedMessage,

    String paddedMessage,

    String decryptedMessage,

    Integer blockSize,

    List<Integer> permutation,

    Integer shift,

    List<AuditStep> audit
){


}

File: frontend/SCMC_Frontend/src/modules/cipher/types/DecryptResponse.ts
import type { AuditStep } from "./AuditStep";

export interface DecryptResponse {

    encryptedMessage: string;

    permutedMessage: string;

    paddedMessage: string;

    decryptedMessage: string;

    blockSize: number;

    permutation: number[];

    shift: number;

    audit: AuditStep[];

}